src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -3766,6 +3776,20 @@ protected IndexQueryDoneRunning CurrentlyInUse()
             return queryDoneRunning;
         }
 
+        private IDisposable EnsureSingleIndexingThread()
+        {
+            try
+            {
+                _executingIndexing.Wait(_indexingProcessCancellationTokenSource.Token);
+            }
+            catch (OperationCanceledException)
+            {
+                return null;
+            }
```

Comment:
What scenario would cause this?

Reply:
It's just defensive code against trying to get the lock while `_indexingProcessCancellationTokenSource.Token` is already cancelled.